openfisca_core/periods.py

```diff
@@ -27,7 +27,7 @@
 YEAR = 'year'
 ETERNITY = 'eternity'
 
-INSTANT_PATTERN = re.compile('^\d{4}(?:-\d{1,2}){0,2}$')  # matches '2015', '2015-01', '2015-01-01'
+INSTANT_PATTERN = re.compile(r'^\d{4}(?:-\d{1,2}){0,2}$')  # matches '2015', '2015-01', '2015-01-01'
```

Comment:
I have no idea why this "r" prefix is needed, why a minor linter update is requiring this change, or what I should do in the future to stay compliant; I'd rather not accept this PR without understanding what's going on. This does not necessarily require a code change, but definitely some explanation in the PR description.